keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md

```diff
@@ -254,12 +254,36 @@ the number if needed, but upgrades aren't that frequent and few resources are
 changed on releases, so these requests should not be common. We will count them
 with a metric.
 
+There could be a large volume of requests for a specific resource which might result in the identified apiserver being unable to serve the proxied requests. This scenario should not occur too frequently, since resource types which have large request volume should not be added or removed during an upgrade -- that would cause other problems, too.
+
 TODO: security / cert stuff.
 
 ## Design Details
 
-TODO: explanation of how the handler will determine a request is for a resource
-that should be proxied.
+Update the handler stack to detect the condition where proxying of the request to an alternate apiserver is needed.
+```
+func (h *handler) ServeHttp(req *http.Request){
+
+  // find the apiserverid that knows of the request's GVR
+  APIServerIdForResource = StorageVersion.getServerIdsForGVR(req))
+  
+  if (!APIServerIdsForResource) {
+    // no apiserver found that knows about req.resource
+    return (HttpResponse(503))
+  }
+  
+  // get network location of the apiserver along with public key (think of where to make this info available)
+  APIServerLocationInfo = APIService.getNetworkLocation(APIServerIdForResource)
+  
+  // save APIserver network location in request context
+  req = req.WithContext(APIServerLocationInfo)
+  
```

Comment:
The 503 comes in if this proxy fails for some reason.

Reply:
Okay but this will happen after the request has been through the entire handler stack and is finally served at some point where it will fail, right?

Reply:
No -- if we proxy, we're cutting short the rest of the filters and handlers, we don't call them at all.

Reply:
So this should happen when the request has passed through the basic set of filters (that make sense to have- like the initial few in the chain specified here https://github.com/kubernetes/kubernetes/blob/d2be69ac11346d2a0fab8c3c168c4255db99c56f/staging/src/k8s.io/apiserver/pkg/server/config.go#L896 adding logging, tracing, retries etc.)

Reply:
Right.